pkg/metrics/status.go

```diff
@@ -154,4 +195,98 @@ func (s *statusCollector) Collect(ch chan<- prometheus.Metric) {
 		prometheus.GaugeValue,
 		float64(unreachableEndpoints),
 	)
+
+	s.collectNodeConnectivityMetrics(ch, healthStatusResponse)
+}
+
+func (s *statusCollector) collectNodeConnectivityMetrics(ch chan<- prometheus.Metric, response *connectivity.GetStatusOK) {
+	if response == nil || response.Payload == nil || len(response.Payload.Nodes) == 0 {
+		return
+	}
+
+	localClusterName, localNodeName := getClusterNodeName(response.Payload.Local.Name)
+
+	for _, n := range response.Payload.Nodes {
+		if n == nil || n.Host == nil || n.Host.PrimaryAddress == nil || n.HealthEndpoint == nil || n.HealthEndpoint.PrimaryAddress == nil {
+			continue
+		}
+
+		targetClusterName, targetNodeName := getClusterNodeName(n.Name)
+		nodePathStatus := healthClientPkg.GetHostPrimaryAddress(n)
+		nodePathConnectivityStatusType := healthClientPkg.GetPathConnectivityStatusType(nodePathStatus)
+		endpointPathStatus := n.HealthEndpoint
+		endpointPathConnectivityStatusType := healthClientPkg.GetPathConnectivityStatusType(endpointPathStatus.PrimaryAddress)
+		isEndpointReachable := endpointPathConnectivityStatusType == healthClientPkg.ConnStatusReachable
+		isNodeReachable := nodePathConnectivityStatusType == healthClientPkg.ConnStatusReachable
+		nodeType := LabelValueLocal
+
+		if targetClusterName != localClusterName {
+			nodeType = LabelValueRemote
+		}
+
+		ch <- prometheus.MustNewConstMetric(s.nodeConnectivityStatusDesc,
+			prometheus.GaugeValue,
+			boolToFloat64(isEndpointReachable),
+			localClusterName, localNodeName, targetClusterName, targetNodeName, nodePathStatus.IP, nodeType, LabelValueEndpoint,
```

Comment:
The IP address seems to be wrong here? Unless this is intentionally trying to group the metrics for both the endpoint + the node itself together without including the actual target IP for the health probe?

```suggestion
			localClusterName, localNodeName, targetClusterName, targetNodeName, endpointPathStatus.IP, nodeType, LabelValueEndpoint,
```

Reply:
you are right, the IP here is wrong :100: